bindings/wasm/src/account/storage/traits.rs

```diff
@@ -0,0 +1,307 @@
+  async fn set_published_generation(&self, did: &IotaDID, index: Generation) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_published_generation(did.clone().into(), index.into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Returns the chain state of the identity specified by `did`.
+  async fn chain_state(&self, did: &IotaDID) -> AccountResult<Option<ChainState>> {
+    let promise: Promise = Promise::resolve(&self.chain_state(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Set the chain state of the identity specified by `did`.
+  async fn set_chain_state(&self, did: &IotaDID, chain_state: &ChainState) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_chain_state(did.clone().into(), chain_state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Returns the state of the identity specified by `did`.
+  async fn state(&self, did: &IotaDID) -> AccountResult<Option<IdentityState>> {
+    let promise: Promise = Promise::resolve(&self.state(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Sets a new state for the identity specified by `did`.
+  async fn set_state(&self, did: &IotaDID, state: &IdentityState) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.set_state(did.clone().into(), state.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+
+  /// Removes the keys and any state for the identity specified by `did`.
+  async fn purge(&self, did: &IotaDID) -> AccountResult<()> {
+    let promise: Promise = Promise::resolve(&self.purge(did.clone().into()));
+    let result: JsValueResult = JsFuture::from(promise).await.into();
+    result.into()
+  }
+}
+
+#[wasm_bindgen(typescript_custom_section)]
+const STORAGE: &'static str = r#"
+/** All methods an object must implement to be used as an account storage. */
+interface Storage {
+  /** Sets the account password.*/
+  setPassword: (encryptionKey: Uint8Array) => Promise<void>;
+
+  /** Write any unsaved changes to disk.*/
+  flushChanges: () => Promise<void>;
+
+  /** 
+   * Attempt to obtain the exclusive permission to modify the given `did`.
+   * The caller is expected to make no more modifications after the lease has been dropped.
+   * Returns an error if already leased.
+  */
+  leaseDid: (did: DID) => Promise<DIDLease>;
+
+  /** Creates a new keypair at the specified `location`.*/
+  keyNew: (did: DID, keyLocation: KeyLocation) => Promise<string>;
+
+  /** Inserts a private key at the specified `location`.*/
+  keyInsert: (did: DID, keyLocation: KeyLocation, privateKey: string) => Promise<string>;
```

Comment:
- All the places that take a key as a string or return one, should document that the (expected) format is base58. 
- The documentation of the storage trait in Rust is also lacking a bit, we should perhaps fix that as well. `key_*` functions that are expected to return a public key should state so in the documentation. This is particularly important for the JS version where a `string` is a less meaningful return type than `PublicKey` in Rust.